exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceClients.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.trace.ocagent;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.MoreExecutors;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.opencensus.proto.agent.common.v1.Node;
+import io.opencensus.proto.agent.trace.v1.CurrentLibraryConfig;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceRequest;
+import io.opencensus.proto.agent.trace.v1.ExportTraceServiceResponse;
+import io.opencensus.proto.agent.trace.v1.TraceServiceGrpc;
+import io.opencensus.proto.agent.trace.v1.TraceServiceGrpc.TraceServiceStub;
+import io.opencensus.proto.agent.trace.v1.UpdatedLibraryConfig;
+import io.opencensus.proto.trace.v1.Span;
+import io.opencensus.trace.export.SpanData;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/** Implementation of OC-Agent Trace Service clients. */
+final class OcAgentTraceServiceClients {
+
+  private OcAgentTraceServiceClients() {}
+
+  private static final Logger logger = Logger.getLogger(OcAgentTraceServiceClients.class.getName());
+  private static final DaemonThreadFactory factory = new DaemonThreadFactory();
+
+  // A reference to the callback currentConfigObserver returned from Agent server.
+  // Shared between main thread and AgentConnectionWorker thread.
+  // The reference is non-null only when the connection succeeded.
+  @VisibleForTesting
+  static final AtomicReference<StreamObserver<CurrentLibraryConfig>> currentConfigObserverRef =
+      new AtomicReference<>();
+
+  // A reference to the callback exportRequestObserver returned from Agent server.
+  // Shared between main thread and AgentConnectionWorker thread.
+  // The reference is non-null only when the connection succeeded.
+  @VisibleForTesting
+  static final AtomicReference<StreamObserver<ExportTraceServiceRequest>> exportRequestObserverRef =
+      new AtomicReference<>();
+
+  private static boolean isExportConnected() {
+    return exportRequestObserverRef.get() != null;
+  }
+
+  private static boolean isConfigConnected() {
```

Comment:
We should also have an option in the configuration of the ocagent exporter "enable_config" or something similar that allows users to configure if this exporter should also handle the config.

Also based on some comments on gitter we may have a configuration to that says keep connection open (for Android devices we may not want to always keep the RPC connection/request alive). We can address this later.